sysdump/sysdump.go

```diff
@@ -0,0 +1,868 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sysdump
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/cilium/cilium-cli/internal/utils"
+	"github.com/cilium/workerpool"
+	"github.com/hashicorp/go-multierror"
+	archiver "github.com/mholt/archiver/v3"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+// SysdumpOptions groups together the set of options required to collect a sysdump.
+type SysdumpOptions struct {
+	// The labels used to target Cilium pods.
+	CiliumLabelSelector string
+	// The namespace Cilium is running in.
+	CiliumNamespace string
+	// The labels used to target Cilium operator pods.
+	CiliumOperatorLabelSelector string
+	// The namespace Cilium operator is running in.
+	CiliumOperatorNamespace string
+	// Whether to enable debug logging.
+	Debug bool
+	// The labels used to target Hubble pods.
+	HubbleLabelSelector string
+	// The namespace Hubble is running in.
+	HubbleNamespace string
+	// The labels used to target Hubble Relay pods.
+	HubbleRelayLabelSelector string
+	// The namespace Hubble Relay is running in.
+	HubbleRelayNamespace string
+	// The labels used to target Hubble UI pods.
+	HubbleUILabelSelector string
+	// The namespace Hubble UI is running in.
+	HubbleUINamespace string
+	// The amount of time to wait for the user to cancel the sysdump on a large cluster.
+	LargeSysdumpAbortTimeout time.Duration
+	// The threshold on the number of nodes present in the cluster that triggers a warning message.
+	LargeSysdumpThreshold int32
+	// The limit on the number of bytes to retrieve when collecting logs
+	LogsLimitBytes int64
+	// How far back in time to go when collecting logs.
+	LogsSinceTime time.Duration
+	// Comma-separated list of node IPs or names to filter pods for which to collect gops and logs.
+	NodeList string
+	// The name of the resulting file (without extension)\n'<ts>' can be used as the placeholder for the timestamp.
+	OutputFileName string
+	// Whether to enable quick mode (i.e. skip collection of 'cilium-bugtool' output and logs).
+	Quick bool
+	// A 'RESTClientGetter' that can be used to create REST clients for the Kubernetes API.
+	// Required at least for getting the proper output of 'kubectl get pod -o wide' without actually using 'kubectl'.
+	RESTClientGetter genericclioptions.RESTClientGetter
+	// The number of workers to use.
+	WorkerCount int32
```

Comment:
I'd stick to using `int` unless there's a good reason to use `int32`. I see that this member is being cast to an `int` when used anyway.